repos/system_upgrade/common/actors/scanpkgmanager/libraries/scanpkgmanager.py

```diff
@@ -28,5 +33,53 @@ def get_etc_releasever():
     return releasever
 
 
+def _get_config_contents(config_path):
+    if os.path.isfile(config_path):
+        with open(config_path, 'r') as config:
+            return config.read()
+    return ''
+
+
+def _get_proxy_if_set(manager_config_path):
+    """
+    Get proxy address from specified package manager config.
+
+    :param str manager_config_path: path to a package manager config
+    :returns: proxy address or None when not set
+    :rtype: str
+    """
+
+    config = _get_config_contents(manager_config_path)
+
+    for line in config.split('\n'):
+        if re.match('^proxy[ \t]*=', line):
+            proxy_address = line.split('=', 1)[1]
+            return proxy_address.strip()
+
+    return None
+
+
+def get_configured_proxies():
+    """
+    Get a list of proxies used in dnf and yum configuration files.
+
+    :returns: list of unique proxies
+    :rtype: List
+    """
+
+    configured_proxies = []
+    for config_path in (DNF_CONFIG_PATH, YUM_CONFIG_PATH):
+        proxy = _get_proxy_if_set(config_path)
+        if proxy and proxy not in configured_proxies:
+            configured_proxies.append(proxy)
+
+    return configured_proxies
```

Comment:
Codewise okay. If you want to collect a unique set of proxies, then just use `set`. Sorting the set when returning results in a list with canonical form, in other words you can predict how will the list look, no matter of the order in what file (dnf/yum conf) was the proxy found. 
```suggestion
    configured_proxies = set()
    for config_path in (DNF_CONFIG_PATH, YUM_CONFIG_PATH):
        proxy = _get_proxy_if_set(config_path)
        if proxy:
            configured_proxies.add(proxy)

    return sorted(configured_proxies)
```